yb-voyager/cmd/analyzeSchema.go

```diff
@@ -676,12 +676,12 @@ func createSqlStrInfoArray(path string, objType string) []sqlInfo {
 
 		// Assuming that both the dollar quote strings will not be in same line
 		if dollarQuoteFlag == 0 {
-			if matches := dollarQuoteRegex.FindStringSubmatch(currLine); matches != nil {
-				dollarQuoteFlag = 1 //denotes start of the code/body part
-				codeBlockDelimiter = matches[0]
-			} else if strings.Contains(currLine, ";") { // in case, there is no body part
+			if strings.Contains(currLine, ";") { // in case, there is no body part or body part is in single line
```

Comment:
What if there is a string literal containing `;` in it? Will it trigger false detection of end of function? How about checking if last non-white-space character is semi-colon?

Reply:
yeah, nice catch. Will make that change.

Reply:
but what if there is a sql comment after the semi-colon